microsoft/PowerToys
# Update check-spelling

## Summary of the Pull Request

**What is this about:**

This improves the patterns used by the spell checker.

**What is include in the PR:** 

Updates to the spell checking engine (it's faster, and it makes more suggestions about things to exclude).

Updates to exclusions (the files were suggested by the updated engine, although I've generalized the `.filters` instance -- you'll want to think about that).

**How does someone test / validate:** 

* Use this commit in your own fork and create PRs into it (especially one that adds a typo).
* or use https://github.com/check-spelling/PowerToys and create a PR into [issue-14854](https://github.com/check-spelling/PowerToys/tree/issue-14854).
* Use [nektos/act](https://github.com/nektos/act) and add a typo locally (just run `act` from the root of your clone).

Note: currently it will include this extra segment in each comment:

  <details><summary>Available dictionaries could cover words not in the dictionary</summary>
  
  [cspell:cpp/cpp.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/cpp/cpp.txt) (104293) covers 208 of them
  [cspell:filetypes/filetypes.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/filetypes/filetypes.txt) (337) covers 39 of them
  [cspell:html/html.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/html/html.txt) (542) covers 28 of them
  [cspell:css/css.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/css/css.txt) (993) covers 27 of them
  [cspell:django/django.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/django/django.txt) (2342) covers 21 of them
  [cspell:aws/aws.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/aws/aws.txt) (1485) covers 19 of them
  [cspell:golang/go.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/golang/go.txt) (7745) covers 18 of them
  [cspell:java/java.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/java/java.txt) (33524) covers 12 of them
  [cspell:csharp/csharp.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/csharp/csharp.txt) (123) covers 11 of them
  [cspell:lua/lua.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/lua/lua.txt) (391) covers 9 of them
  [cspell:fullstack/fullstack.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/fullstack/fullstack.txt) (181) covers 8 of them
  [cspell:scala/scala.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/scala/scala.txt) (2752) covers 7 of them
  [cspell:rust/rust.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/rust/rust.txt) (112) covers 7 of them
  [cspell:python/python.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/python/python.txt) (364) covers 7 of them
  [cspell:npm/npm.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/npm/npm.txt) (671) covers 7 of them
  [cspell:ruby/ruby.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/ruby/ruby.txt) (354) covers 4 of them
  [cspell:php/php.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/php/php.txt) (9785) covers 4 of them
  [cspell:dotnet/dotnet.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/dotnet/dotnet.txt) (9824) covers 3 of them
  [cspell:node/node.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/node/node.txt) (9611) covers 1 of them
  [cspell:bash/bash-words.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/bash/bash-words.txt) (22) covers 1 of them
  [cspell:ada/ada.txt](https://raw.githubusercontent.com/check-spelling/cspell-dicts/HEAD/dictionaries/ada/ada.txt) (72) covers 1 of them
  
  Consider adding them using:
  ```
        with:
          extra_dictionaries:
            cspell:cpp/cpp.txt
            cspell:filetypes/filetypes.txt
            cspell:html/html.txt
            cspell:css/css.txt
            cspell:django/django.txt
            cspell:aws/aws.txt
            cspell:golang/go.txt
            cspell:java/java.txt
            cspell:csharp/csharp.txt
            cspell:lua/lua.txt
            cspell:fullstack/fullstack.txt
            cspell:scala/scala.txt
            cspell:rust/rust.txt
            cspell:python/python.txt
            cspell:npm/npm.txt
            cspell:ruby/ruby.txt
            cspell:php/php.txt
            cspell:dotnet/dotnet.txt
            cspell:node/node.txt
            cspell:bash/bash-words.txt
            cspell:ada/ada.txt
  ```
  To stop checking additional dictionaries, add:
  ```
        with:
          check_extra_dictionaries: ''
  ```
  
  </details>

It might make sense to add the following to the spelling workflow:

```
        with:
          extra_dictionaries:
            cspell:cpp/cpp.txt
            cspell:filetypes/filetypes.txt
            cspell:html/html.txt
            cspell:css/css.txt
```
or
```
        with:
          extra_dictionaries:
            cspell:filetypes/filetypes.txt
            cspell:html/html.txt
            cspell:css/css.txt
```

The inclusion of `cpp.txt` is questionable as unfortunately it has quite a few typos included (I haven't had the energy to edit the file to remove them), so this would unfortunately mean the spell checker would skip flagging some actual typos. I think that `filetypes.txt`, `html.txt`, and `css.txt` would be fairly reasonable.

Alternatively, it might be best to just go with:
```
        with:
          check_extra_dictionaries: ''
```
which turns off the intelligence entirely.

## Quality Checklist

- [x] **Linked issue:** #14854

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
